### ISSUE TITLE
[Bug]: vllm using ray in eks hangs when using --pipeline_parallel_size > 1

### ISSUE THREAD

**Issue by ran-haim** [vllm-project/vllm#11139]
### Your current environment

running on a pod in g6.12xlarge (allocated by lws).
Pod is initializing ray before running vllm (using the proposed lws image https://github.com/kubernetes-sigs/lws/blob/main/docs/examples/vllm/build/Dockerfile.GPU)

### 🐛 Describe the bug

Vllm is stuck on this meesage:
INFO 12-12 05:28:31 pynccl.py:69] vLLM is using nccl==2.21.5

full log:
[2024-12-12 05:27:53,632 W 8 8] global_state_accessor.cc:463: Retrying to get node with node ID 9c36d691ad808fe6b12015dc3c0c4ba0432917a72547d5450434659c
2024-12-12 05:27:52,822 INFO usage_lib.py:467 -- Usage stats collection is enabled by default without user confirmation because this terminal is detected t
o be non-interactive. To disable this, add `--disable-usage-stats` to the command that starts the cluster, or run the following command: `ray disable-usage
-stats` before starting the cluster. See https://docs.ray.io/en/master/cluster/usage-stats.html for more details.
2024-12-12 05:27:52,822 INFO scripts.py:822 -- Local node IP: 10.0.143.175
2024-12-12 05:27:54,657 SUCC scripts.py:859 -- --------------------
2024-12-12 05:27:54,657 SUCC scripts.py:860 -- Ray runtime started.
2024-12-12 05:27:54,657 SUCC scripts.py:861 -- --------------------
2024-12-12 05:27:54,657 INFO scripts.py:863 -- Next steps
2024-12-12 05:27:54,657 INFO scripts.py:866 -- To add another node to this Ray cluster, run
2024-12-12 05:27:54,657 INFO scripts.py:869 --   ray start --address='10.0.143.175:6379'
2024-12-12 05:27:54,657 INFO scripts.py:878 -- To connect to this Ray cluster:
2024-12-12 05:27:54,657 INFO scripts.py:880 -- import ray
2024-12-12 05:27:54,657 INFO scripts.py:881 -- ray.init()
2024-12-12 05:27:54,657 INFO scripts.py:912 -- To terminate the Ray runtime, run
2024-12-12 05:27:54,657 INFO scripts.py:913 --   ray stop
2024-12-12 05:27:54,657 INFO scripts.py:916 -- To view the status of the cluster, use
2024-12-12 05:27:54,657 INFO scripts.py:917 --   ray status
2024-12-12 05:27:55,002 INFO worker.py:1634 -- Connecting to existing Ray cluster at address: 10.0.143.175:6379...
2024-12-12 05:27:55,010 INFO worker.py:1819 -- Connected to Ray cluster.
All ray workers are active and the ray cluster is initialized successfully.
INFO 12-12 05:27:59 api_server.py:585] vLLM API server version 0.6.4.post1
INFO 12-12 05:27:59 api_server.py:586] args: Namespace(host=None, port=8080, uvicorn_log_level='info', allow_credentials=False, allowed_origins=['*'], allo
wed_methods=['*'], allowed_headers=['*'], api_key=None, lora_modules=None, prompt_adapters=None, chat_template=None, response_role='assistant', ssl_keyfile
=None, ssl_certfile=None, ssl_ca_certs=None, ssl_cert_reqs=0, root_path=None, middleware=[], return_tokens_as_token_ids=False, disable_frontend_multiproces
sing=False, enable_auto_tool_choice=False, tool_call_parser=None, tool_parser_plugin='', model='meta-llama/Llama-3.1-8B-Instruct', task='auto', tokenizer=N
one, skip_tokenizer_init=False, revision=None, code_revision=None, tokenizer_revision=None, tokenizer_mode='auto', chat_template_text_format='string', trus
t_remote_code=False, allowed_local_media_path=None, download_dir=None, load_format='auto', config_format=<ConfigFormat.AUTO: 'auto'>, dtype='auto', kv_cach
e_dtype='auto', quantization_param_path=None, max_model_len=None, guided_decoding_backend='outlines', distributed_executor_backend=None, worker_use_ray=Fal
se, pipeline_parallel_size=2, tensor_parallel_size=4, max_parallel_loading_workers=None, ray_workers_use_nsight=False, block_size=16, enable_prefix_caching
=False, disable_sliding_window=False, use_v2_block_manager=False, num_lookahead_slots=0, seed=0, swap_space=4, cpu_offload_gb=0, gpu_memory_utilization=0.9
, num_gpu_blocks_override=None, max_num_batched_tokens=None, max_num_seqs=256, max_logprobs=20, disable_log_stats=False, quantization=None, rope_scaling=No
ne, rope_theta=None, hf_overrides=None, enforce_eager=False, max_seq_len_to_capture=8192, disable_custom_all_reduce=False, tokenizer_pool_size=0, tokenizer
_pool_type='ray', tokenizer_pool_extra_config=None, limit_mm_per_prompt=None, mm_processor_kwargs=None, enable_lora=False, enable_lora_bias=False, max_lora
s=1, max_lora_rank=16, lora_extra_vocab_size=256, lora_dtype='auto', long_lora_scaling_factors=None, max_cpu_loras=None, fully_sharded_loras=False, enable_
prompt_adapter=False, max_prompt_adapters=1, max_prompt_adapter_token=0, device='auto', num_scheduler_steps=1, multi_step_stream_outputs=True, scheduler_de
lay_factor=0.0, enable_chunked_prefill=None, speculative_model=None, speculative_model_quantization=None, num_speculative_tokens=None, speculative_disable_
mqa_scorer=False, speculative_draft_tensor_parallel_size=None, speculative_max_model_len=None, speculative_disable_by_batch_size=None, ngram_prompt_lookup_
max=None, ngram_prompt_lookup_min=None, spec_decoding_acceptance_method='rejection_sampler', typical_acceptance_sampler_posterior_threshold=None, typical_a
cceptance_sampler_posterior_alpha=None, disable_logprobs_during_spec_decoding=None, model_loader_extra_config=None, ignore_patterns=[], preemption_mode=Non
e, served_model_name=None, qlora_adapter_name_or_path=None, otlp_traces_endpoint=None, collect_detailed_traces=None, disable_async_output_proc=False, sched
uling_policy='fcfs', override_neuron_config=None, override_pooler_config=None, disable_log_requests=False, max_log_len=None, disable_fastapi_docs=False, en
able_prompt_tokens_details=False)
INFO 12-12 05:28:04 config.py:350] This model supports multiple tasks: {'generate', 'embedding'}. Defaulting to 'generate'.
INFO 12-12 05:28:04 config.py:1020] Defaulting to use ray for distributed inference
WARNING 12-12 05:28:04 arg_utils.py:1013] Chunked prefill is enabled by default for models with max_model_len > 32K. Currently, chunked prefill might not w
ork with some features or models. If you encounter any issues, please disable chunked prefill by setting --enable-chunked-prefill=False.
WARNING 12-12 05:28:04 arg_utils.py:1075] [DEPRECATED] Block manager v1 has been removed, and setting --use-v2-block-manager to True or False has no effect
 on vLLM behavior. Please remove --use-v2-block-manager in your engine argument. If your use case is not supported by SelfAttnBlockSpaceManager (i.e. block
 manager v2), please file an issue with detailed information.
INFO 12-12 05:28:04 config.py:1136] Chunked prefill is enabled with max_num_batched_tokens=512.
WARNING 12-12 05:28:04 config.py:479] Async output processing can not be enabled with pipeline parallel
2024-12-12 05:28:04,536 INFO worker.py:1634 -- Connecting to existing Ray cluster at address: 10.0.143.175:6379...
2024-12-12 05:28:04,543 INFO worker.py:1819 -- Connected to Ray cluster.
INFO 12-12 05:28:04 llm_engine.py:249] Initializing an LLM engine (v0.6.4.post1) with config: model='meta-llama/Llama-3.1-8B-Instruct', speculative_config=
None, tokenizer='meta-llama/Llama-3.1-8B-Instruct', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config=None, tokenizer_r
evision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=131072, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=4, pi
peline_parallel_size=2, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto, quantization_param_path=None, device_
config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='outlines'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None, collec
t_model_forward_time=False, collect_model_execute_time=False), seed=0, served_model_name=meta-llama/Llama-3.1-8B-Instruct, num_scheduler_steps=1, chunked_p
refill_enabled=True multi_step_stream_outputs=True, enable_prefix_caching=False, use_async_output_proc=False, use_cached_outputs=False, chat_template_text_
format=string, mm_processor_kwargs=None, pooler_config=None)
INFO 12-12 05:28:06 ray_gpu_executor.py:134] use_ray_spmd_worker: False
INFO 12-12 05:28:29 selector.py:135] Using Flash Attention backend.
←[36m(RayWorkerWrapper pid=365)←[0m INFO 12-12 05:28:29 selector.py:135] Using Flash Attention backend.
INFO 12-12 05:28:31 utils.py:961] Found nccl from library libnccl.so.2
INFO 12-12 05:28:31 pynccl.py:69] vLLM is using nccl==2.21.5
vllm-0:936:936 [0] NCCL INFO Bootstrap : Using eth0:10.0.143.175<0>
vllm-0:936:936 [0] NCCL INFO NET/Plugin: No plugin found (libnccl-net.so)
vllm-0:936:936 [0] NCCL INFO NET/Plugin: Plugin load returned 2 : libnccl-net.so: cannot open shared object file: No such file or directory : when loading
libnccl-net.so
vllm-0:936:936 [0] NCCL INFO NET/Plugin: Using internal network plugin.
vllm-0:936:936 [0] NCCL INFO cudaDriverVersion 12040
NCCL version 2.21.5+cuda12.4
vllm-0:936:936 [0] NCCL INFO NET/IB : No device found.
vllm-0:936:936 [0] NCCL INFO NET/Socket : Using [0]eth0:10.0.143.175<0>
vllm-0:936:936 [0] NCCL INFO Using non-device net plugin version 0
vllm-0:936:936 [0] NCCL INFO Using network Socket
vllm-0:936:936 [0] NCCL INFO ncclCommInitRank comm 0x31f8ebd0 rank 0 nranks 4 cudaDev 0 nvmlDev 0 busId 38000 commId 0x9419ffc51c422e05 - Init START
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO Setting affinity for GPU 0 to ffff,ffffffff
vllm-0:936:936 [0] NCCL INFO NVLS multicast support is not available on dev 0
vllm-0:936:936 [0] NCCL INFO comm 0x31f8ebd0 rank 0 nRanks 4 nNodes 1 localRanks 4 localRank 0 MNNVL 0
vllm-0:936:936 [0] NCCL INFO Channel 00/02 :    0   1   2   3
vllm-0:936:936 [0] NCCL INFO Channel 01/02 :    0   1   2   3
vllm-0:936:936 [0] NCCL INFO Trees [0] 1/-1/-1->0->-1 [1] 1/-1/-1->0->-1
vllm-0:936:936 [0] NCCL INFO P2P Chunksize set to 131072
vllm-0:936:936 [0] NCCL INFO NCCL_CUMEM_ENABLE set by environment to 0.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGNORE_DISABLED_P2P=1.
vllm-0:936:936 [0] NCCL INFO Channel 00 : 0[0] -> 1[1] via SHM/direct/direct
vllm-0:936:936 [0] NCCL INFO P2P is disabled between ←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m INFO 12-12 05:28:31 utils.py:961] Found nccl from
 library libnccl.so.2
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m INFO 12-12 05:28:31 pynccl.py:69] vLLM is using nccl==2.21.5
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vllm-0-1:194:194 [0] NCCL INFO Bootstrap : Using eth0:10.0.130.237<0>
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vllm-0-1:194:194 [0] NCCL INFO NET/Plugin: No plugin found (libnccl-net.so)
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vllm-0-1:194:194 [0] NCCL INFO NET/Plugin: Plugin load returned 2 : libnccl-net.so: cannot open shared
 object file: No such file or directory : when loading libnccl-net.so
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vllm-0-1:194:194 [0] NCCL INFO NET/Plugin: Using internal network plugin.
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vllm-0-1:194:194 [0] NCCL INFO cudaDriverVersion 12040
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m NCCL version 2.21.5+cuda12.4
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO NET/IB : No device found.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO NET/Socket : Using [0]eth0:10.0.143.175<0>
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO Using non-device net plugin version 0
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO Using network Socket
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO ncclCommInitRank comm 0x2c7cef60 rank 1 nranks 4 cudaDev 1 nvmlDev 1 busId 3a000 commId 0x
9419ffc51c422e05 - Init START
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 1. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 0 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 2 and 3. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO Setting affinity for GPU 1 to ffff,ffffffff
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO NVLS multicast support is not available on dev 1
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO comm 0x2c7cef60 rank 1 nRanks 4 nNodes 1 localRanks 4 localRank 1 MNNVL 0
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO Trees [0] 2/-1/-1->1->0 [1] 2/-1/-1->1->0
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P Chunksize set to 131072
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO NCCL_CUMEM_ENABLE set by environment to 0.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this message with NCCL_IGN
ORE_DISABLED_P2P=1.
←[36m(RayWorkerWrapper pid=367)←[0m vllm-0:367:367 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this
←[36m(RayWorkerWrapper pid=365)←[0m vllm-0:365:365 [3] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress th
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vllm-0-1:194:194 [0] NCCL INFO Channel 00/02 :    0   1   2   3
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vllm-0-1:194:194 [0] NCCL INFO Channel 01/02 :    0   1   2   3
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m vl
←[36m(RayWorkerWrapper pid=264, ip=10.0.130.237)←[0m vllm-0-1:264:264 [1] NCCL INFO P2P is disabled between connected GPUs 1 and 0. You can repress this me
ssage
←[36m(RayWorkerWrapper pid=335, ip=10.0.130.237)←[0m vllm-0-1:335:335 [2] NCCL INFO P2P is disabled between connected GPUs 2 and 1. You can repress this me
ssage wi
←[36m(RayWorkerWrapper pid=406, ip=10.0.130.237)←[0m vllm-0-1:406:406 [3] NCCL INFO P2P is disabled between connected GPUs 3 and 2. You can repress this me
ssag
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m WARNING 12-12 05:28:31 custom_all_reduce.py:134] Custom allreduce is disabled because it's not support
ed on more than two PCIe-only GPUs. To silence this warning, specify disable_custom_all_reduce=True explicitly.
←[36m(RayWorkerWrapper pid=194, ip=10.0.130.237)←[0m INFO 12-12 05:28:31 shm_broadcast.py:236] vLLM message queue communication handle: Handle(connect_ip='
127.0.0.1', local_reader_ranks=[1, 2, 3], buffer=<vllm.distributed.device_communicators.shm_broadcast.ShmRingBuffer object at 0x7efce816d8b0>, local_subscr
ibe_port=44055, remote_subscribe_port=None)
WARNING 12-12 05:28:31 custom_all_reduce.py:134] Custom allreduce is disabled because it's not supported on more than two PCIe-only GPUs. To silence this w
arning, specify disable_custom_all_reduce=True explicitly.
INFO 12-12 05:28:31 shm_broadcast.py:236] vLLM message queue communication handle: Handle(connect_ip='127.0.0.1', local_reader_ranks=[1, 2, 3], buffer=<vll
m.distributed.device_communicators.shm_broadcast.ShmRingBuffer object at 0x7feb60210b90>, local_subscribe_port=33777, remote_subscribe_port=None)
INFO 12-12 05:28:31 utils.py:961] Found nccl from library libnccl.so.2
INFO 12-12 05:28:31 pynccl.py:69] vLLM is using nccl==2.21.5

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by ran-haim** [comment]
no one encountered that?

**Comment by ruisearch42** [comment]
Hi @ran-haim , what happens if you use multi-processing backend. Do you see the same issue?

**Comment by ran-haim** [comment]
Hi,
I am not sure how to set that up, are there docs for that?

**Comment by ran-haim** [comment]
Do you mean running on one node without ray?
Yes it works without issue.

**Comment by ran-haim** [comment]
did someone ever manage to make this work somehow?

**Comment by bryantbiggs** [comment]
this issue should have the `aws-neuron` label FYI

I didn't see details on your cluster setup, but just in case, are you familiar with when you need to use the Neuron scheduler extension for Kubernetes? https://awsdocs-neuron.readthedocs-hosted.com/en/latest/containers/tutorials/k8s-neuron-scheduler.html

**Comment by ran-haim** [comment]
why neuron? 
I am using nvidia gpu.

**Comment by ran-haim** [comment]
I am using istio, and it seems like istio-proxy is causing the hang.
though I have no idea what exactly is wrong, as grpc between the nodes is working fine - most of the task end successfully.
one task stays on executing forever, I do not see any error or anything else - I do think it is running on the node ncclCommInitRank (at least that's what I am seeing in the GUI).

**Comment by bryantbiggs** [comment]
Apologies, wrong issue - meant to comment on the one with inf2
